js/components/portal-dashboard/legend-container.tsx

```diff
@@ -1,12 +1,12 @@
 import React from "react";
 
-import css from "../../../css/portal-dashboard/progress-legend.less";
+import css from "../../../css/portal-dashboard/progress-view-legend.less";
 
-export interface ProgressType {
-  name: string;
+export interface LegendType {
+  name: string | null;
```

Comment:
You could probably also write this simply as:
`name?: string;`

Reply:
While that suggestion may be correct, it's not equivalent given that `null` !== `undefined`.

Reply:
Agreed, not equivalent, but I'm suggesting the removal of null entirely in the comment below.  We will accomplish the same thing since this code is just trying to determine if there is a name or not (the use of null doesn't do anything unique from using undefined in this case).